src/lib.rs

```diff
@@ -536,28 +536,19 @@ impl LightningNode {
         }
     }
 
-    // TODO: restrict returned payments according to `number_of_payments` parameter
-    pub fn get_latest_payments(&self, _number_of_payments: u32) -> Result<Vec<Payment>> {
-        let breez_payments = self
-            .rt
+    pub fn get_latest_payments(&self, number_of_payments: u32) -> Result<Vec<Payment>> {
+        self.rt
             .handle()
             .block_on(self.sdk.list_payments(PaymentTypeFilter::All, None, None))
             .map_to_runtime_error(
                 RuntimeErrorCode::NodeUnavailable,
                 "Failed to get payment by hash",
```

Comment:
This problem was already here, but since you're already here, could you fix the error message? :) 

```suggestion
                "Failed to list payments",
```